server/user.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * @fileoverview user function handler
+ */
+var fs = require('fs');
+var assert = require('assert');
+
+var log = require('./log.js');
+var utils = require('./utils.js');
+
+/** @type {user} */
+module.exports = user;
+
+/**
+ * @constructor
+ */
+function user() {}
+
+/** @enum {string} */
+user.QueryType = {
+  SIGNUP: 'sign-up',
+  SIGNIN: 'sign-in'
+};
+
+/**
+ * Name of user information file.
+ * @type {string}
+ */
+user.userInfoFile = 'userInfo.txt';
+
+/**
+ * Expire time of cookie.
+ * @const {number}
+ */
+user.cookieExpireTime = 24 * 60 * 60 * 1000;
+
+/**
+ * @typedef {{
+ *   email: (string|undefined),
+ *   username: string,
+ *   password: string,
+ *   confirmed: (boolean|undefined)
+ * }}
+ */
+user.userInfo;
+
+/**
+ * @typedef {{
+ *   error: (Object<{
+ *     type: string,
+ *     message: string
+ *   }>)
+ * }}
+ */
+user.Error;
+
+/**
+ * Checks the user information for signing in.
+ * @param {string} userPath Directory of user indormation file.
+ * @param {!user.userInfo} userInfo User Information
+ * @return {user.Error|boolean}
+ */
+user.signUp = function(userPath, userInfo) {
```

Comment:
I don't think server-side validation is necessary. Validation is used to ensure user's input are valid. If not, we should remind user that your input are invalid please retype. That is why I think validation should be held in front end but not server.